src/main/java/org/folio/service/pieces/flows/update/PieceUpdateFlowManager.java

```diff
@@ -92,10 +111,49 @@ public Future<Void> updatePiece(Piece pieceToUpdate, boolean createItem, boolean
   }
 
   protected Future<Void> updatePoLine(PieceUpdateHolder holder, RequestContext requestContext) {
-    if (!Boolean.TRUE.equals(holder.getOriginPoLine().getIsPackage()) && !Boolean.TRUE.equals(holder.getOriginPoLine().getCheckinItems())) {
-      return updatePoLineService.updatePoLine(holder, requestContext);
-    }
-    return Future.succeededFuture();
+    CompositePoLine originPoLine = holder.getOriginPoLine();
+
+    return getPiecesByPoLine(originPoLine.getId(), requestContext)
+      .map(pieces -> {
+        CompositePurchaseOrder order = holder.getOriginPurchaseOrder();
+        if (order.getOrderType() != OrderType.ONE_TIME || order.getWorkflowStatus() != WorkflowStatus.OPEN) {
+          return Future.succeededFuture();
+        }
```

Comment:
Better to move these lines before invocation getPiecesByPoLine

Reply:
But I don't see a nice way to compose with following logic. Can you show a code?

Reply:
sure: 
`  protected Future<Void> updatePoLine(PieceUpdateHolder holder, RequestContext requestContext) {
    CompositePoLine originPoLine = holder.getOriginPoLine();
    CompositePurchaseOrder order = holder.getOriginPurchaseOrder();

    if (order.getOrderType() != OrderType.ONE_TIME || order.getWorkflowStatus() != WorkflowStatus.OPEN) {
      return Future.succeededFuture();
    }

    return pieceStorageService.getPiecesByLineId(originPoLine.getId(), requestContext)
      .map(pieces -> {
        List<Piece> piecesToUpdate = List.of(holder.getPieceToUpdate());
        CompositePoLine poLineToSave = holder.getPoLineToSave();
        return poLineToSave.withReceiptStatus(calculatePoLineReceiptStatus(originPoLine, pieces, piecesToUpdate));
      })
      .compose(poLineToSave -> purchaseOrderLineService.saveOrderLine(poLineToSave, requestContext))
      .compose(t -> {
        if (!Boolean.TRUE.equals(originPoLine.getIsPackage()) &&
          !Boolean.TRUE.equals(originPoLine.getCheckinItems())) {
          return updatePoLineService.updatePoLine(holder, requestContext);
        }
        return Future.succeededFuture();
      });
  }`

Reply:
Unfortuntely this won't work because even if order type is not `ONE_TIME` we might need to execute the third compose: last part of the method.